contrib/nitrpg/src/statistics.nit

```diff
@@ -78,6 +69,97 @@ redef class Player
 	end
 end
 
+redef class Counter[E]
+	redef fun inc(e) do
+		self[e] = self[e] + 1
```

Comment:
it calls `self.map[e]` only for the assignement, not the access that goes trough `self[e]`
It comes from the original implementation that did not have a `self[]=` method (only inc and [])